src/cpp/src/whisper_pipeline.cpp

```diff
@@ -28,21 +29,52 @@ ov::genai::OptionalWhisperGenerationConfig get_config_from_map(const ov::AnyMap&
     }
 }
 
-ov::genai::ChunkStreamerVariant get_chunk_streamer_from_map(const ov::AnyMap& config_map) {
-    ov::genai::ChunkStreamerVariant streamer = std::monostate();
-
-    if (config_map.count(ov::genai::utils::STREAMER_ARG_NAME)) {
-        auto any_val = config_map.at(ov::genai::utils::STREAMER_ARG_NAME);
-        if (any_val.is<std::shared_ptr<ov::genai::ChunkStreamerBase>>()) {
-            streamer = any_val.as<std::shared_ptr<ov::genai::ChunkStreamerBase>>();
-        } else if (any_val.is<std::function<bool(std::string)>>()) {
-            streamer = any_val.as<std::function<bool(std::string)>>();
-        } else if (any_val.is<std::function<ov::genai::StreamingStatus(std::string)>>()) {
-            streamer = any_val.as<std::function<ov::genai::StreamingStatus(std::string)>>();
-        }
+ov::genai::StreamerVariant get_streamer_from_map(const ov::AnyMap& config_map) {
+    ov::genai::StreamerVariant streamer = std::monostate();
+
+    if (!config_map.count(ov::genai::utils::STREAMER_ARG_NAME)) {
+        return streamer;
+    }
+
+    auto any_val = config_map.at(ov::genai::utils::STREAMER_ARG_NAME);
+
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    if (any_val.is<std::shared_ptr<ov::genai::StreamerBase>>()) {
+        streamer = any_val.as<std::shared_ptr<ov::genai::StreamerBase>>();
+    } else if (any_val.is<std::function<bool(std::string)>>()) {
+        streamer = any_val.as<std::function<bool(std::string)>>();
+    } else if (any_val.is<std::function<ov::genai::StreamingStatus(std::string)>>()) {
+        streamer = any_val.as<std::function<ov::genai::StreamingStatus(std::string)>>();
     }
+    OPENVINO_SUPPRESS_DEPRECATED_END
+
     return streamer;
 }
+
+std::shared_ptr<ov::genai::StreamerBase> to_base_streamer(ov::genai::StreamerVariant streamer,
+                                                          ov::genai::Tokenizer& tokenizer) {
+    std::shared_ptr<ov::genai::StreamerBase> streamer_ptr;
+
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    if (auto streamer_obj = std::get_if<std::monostate>(&streamer)) {
+        streamer_ptr = nullptr;
+    } else if (auto streamer_obj = std::get_if<std::shared_ptr<ov::genai::StreamerBase>>(&streamer)) {
+        if (auto chunk_streamer = std::dynamic_pointer_cast<ov::genai::ChunkStreamerBase>(*streamer_obj)) {
+            ov::genai::Logger::warn(
+                "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.");
```

Comment:
I think for C++ we warn users via deprecation of `ChunkStreamerBase `
For python users, we need to add message to Pybind11 code.

IMO, in current case C++ code should not print messages.

Reply:
I removed `Logger::warn` added runtime check for pybind code instead: https://github.com/openvinotoolkit/openvino.genai/pull/1769/files#diff-2a65a098c77bed32e5e37a3f588f689464d9b8f5b952c96eef50422655e66149R346
I have an issue with class deprecation - I got no warning for sample (client) code. For genai code warnings appear as expected. Maybe I'm doing it wrong: https://github.com/openvinotoolkit/openvino.genai/pull/1769/files#diff-57b99dea7bc9b9a0b5b94b77a150999f0be5aefd4f9ce0bfd8c758fbaf8b2de9R25

Reply:
hm, looks like gcc warns only about class itself, but when you have a derived class like `ConstructableChunkStreamer` it allows such construction. 

So, it seems that we also need to deprecate all methods of deprecated class as you had initially.

Reply:
I overridden and deprecated all `ChunkStreamerBase` methods. I also added default implementation in the way `StreamerBase` does. That allowed to remove chunk->base streamer adapter.
I still got no deprecation warnings in the sample code though.